library/src/main/java/com/alamkanak/weekview/WeekViewEvent.java

```diff
@@ -101,22 +158,60 @@ public WeekViewEvent(long id, String name, String location, Calendar startTime,
      * @param location The location of the event.
      * @param startTime The time when the event starts.
      * @param endTime The time when the event ends.
+     * @param allDay Is the event an all day event
      */
-    public WeekViewEvent(long id, String name, String location, Calendar startTime, Calendar endTime) {
+    @Deprecated
+    public WeekViewEvent(long id, String name, String location, Calendar startTime, Calendar endTime, boolean allDay) {
+        this(Integer.toString((int) id), name, location, startTime, endTime, allDay, null);
+    }
+
+    /**
+     * Initializes the event for week view.
+     * @param id The id of the event as String.
+     * @param name Name of the event.
+     * @param location The location of the event.
+     * @param startTime The time when the event starts.
+     * @param endTime The time when the event ends.
+     */
+    public WeekViewEvent(String id, String name, String location, Calendar startTime, Calendar endTime) {
         this(id, name, location, startTime, endTime, false);
     }
 
     /**
      * Initializes the event for week view.
      * @param id The id of the event.
      * @param name Name of the event.
+     * @param location The location of the event.
      * @param startTime The time when the event starts.
      * @param endTime The time when the event ends.
      */
-    public WeekViewEvent(long id, String name, Calendar startTime, Calendar endTime) {
+    @Deprecated
+    public WeekViewEvent(long id, String name, String location, Calendar startTime, Calendar endTime) {
+        this(Integer.toString((int) id), name, location, startTime, endTime, false);
+    }
+
+    /**
+     * Initializes the event for week view.
+     * @param id The id of the event specified as String.
+     * @param name Name of the event.
+     * @param startTime The time when the event starts.
+     * @param endTime The time when the event ends.
+     */
+    public WeekViewEvent(String id, String name, Calendar startTime, Calendar endTime) {
         this(id, name, null, startTime, endTime);
     }
 
+    /**
+     * Initializes the event for week view.
+     * @param id The id of the event.
+     * @param name Name of the event.
+     * @param startTime The time when the event starts.
+     * @param endTime The time when the event ends.
+     */
+    @Deprecated
+    public WeekViewEvent(long id, String name, Calendar startTime, Calendar endTime) {
+        this(Integer.toString((int) id), name, null, startTime, endTime);
```

Comment:
Could also call `public WeekViewEvent(long id, String name, String location, Calendar startTime, Calendar endTime, boolean allDay)`, the conversion to int would happen later